.github/workflows/ci.yml

```diff
@@ -72,3 +72,42 @@ jobs:
         run: ruff check .
       - name: Run ruff format
         run: ruff format --check 
+  run-e2e-tests:
+    name: Run E2E tests
+    needs: build-dists
+    permissions:
+      id-token: write
+      contents: read
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [ "3.9", "3.10", "3.11", "3.12", "3.13" ]
+        os: [ ubuntu-latest, macOS-latest, windows-latest ]
+    if: ${{ (github.event_name == 'push' && github.ref == 'refs/heads/main') || github.event_name == 'workflow_dispatch' }}
+    steps:
+      - uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332
+      - name: Download distributions
+        uses: actions/download-artifact@fa0a91b85d4f404e444e00e005971372dc801d16
+        with:
+          name: dists
+          path: dist/
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@39cd14951b08e74b54015e9e001cdefcf80e669f
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install -r requirements-dev.txt
```

Comment:
Let's remove the line for installing from `requirements-dev.txt`. Specifically, for end to end tests, we want to install it as close as we can to what it would be like if installing directly from PyPI, which likely means unpinned dependencies. So, by removing this line, it will make sure we can install the library using the latest versions of our dependencies.